src/rust/cryptography-x509/src/common.rs

```diff
@@ -235,6 +248,28 @@ pub struct DHParams<'a> {
     pub g: asn1::BigUint<'a>,
     pub q: Option<asn1::BigUint<'a>>,
 }
+
+// From PKCS#3 Section 9
+// DHParameter ::= SEQUENCE {
+//     prime INTEGER, -- p
+//     base INTEGER, -- g
+//     privateValueLength INTEGER OPTIONAL
+// }
+#[derive(asn1::Asn1Read, asn1::Asn1Write, Clone, PartialEq, Eq, Debug, Hash)]
+pub struct BasicDHParams<'a> {
+    pub p: asn1::BigUint<'a>,
+    pub g: asn1::BigUint<'a>,
+    pub private_value_length: Option<u32>,
+}
+
+// Based on OpenSSL's `DHxparams`
+#[derive(asn1::Asn1Read, asn1::Asn1Write, Clone, PartialEq, Eq, Debug, Hash)]
+pub struct DHXParams<'a> {
+    pub p: asn1::BigUint<'a>,
+    pub g: asn1::BigUint<'a>,
+    pub q: asn1::BigUint<'a>,
```

Comment:
First, thank you! I guess I'll add j and `validation_params`, even though there's no evidence of their use.

How are you finding the sources for these ASN.1 definitions? It looked to me that this was defined in an ANSI spec and I decided I didn't want to spend 100 francs.

Reply:
I haven't seen a single test vector using either of the extension fields, so I'm going to ignore them for now :-)

Reply:
Your comments have been super helpful here! Can I ask if you have a particular interest in this PR, or just being helpful :-)

Reply:
Pure curiosity what you needed the Brainpool NIDs for. Then I saw the comment indicating a potential gap, and since I happened to know the answer I pointed that out, then I saw that you wanted to document where the ASN.1 came from so I thought I could figure that out as well. That's all.

Reply:
Much appreciated!